openwisp_notifications/api/views.py

```diff
@@ -49,6 +49,8 @@ class BaseNotificationView(GenericAPIView):
     serializer_class = serializers.Serializer
 
     def get_queryset(self):
+        if self.request.user.is_anonymous:
+            return
```

Comment:
@nemesisdesign I don't think these changes makes much logical sense. The permission_classes of this view contains `IsAuthenticated`. Therefore, this view will not be visible to `AnonymousUser`. 

Regarding swagger issues, it only allows unauthenticated user to open API docs when `DEBUG` is set to True. Since that only occurs in the test project and not in deployments (ansible-openwisp2), we should consider updating settings of the project to ignore that issue. 

https://github.com/openwisp/openwisp-utils/blob/4d9e83ae44ae7a82276fae2ceaf2a82ed99ffa5c/openwisp_utils/api/urls.py#L11-L17